frontend/src/components/common/button/button.tsx

```diff
@@ -1,36 +1,60 @@
 import { AppRoute } from 'common/enums/enums';
-import { FC } from 'common/types/types';
+import { FC, IconName } from 'common/types/types';
 import { getValidClasses } from 'helpers/helpers';
+import { ReactElement } from 'react';
 
-import { Link } from '../common';
+import { Icon, Link } from '../common';
 import styles from './styles.module.scss';
 
 type Props = {
   label: string;
+  hasVisuallyHiddenLabel?: boolean;
   btnColor?: 'blue' | 'gray';
   type?: 'button' | 'submit';
-  btnType?: 'filled' | 'outlined';
-  to?: AppRoute;
+  btnType?: 'filled' | 'outlined' | 'icon';
+  to?: AppRoute | string;
```

Comment:
```suggestion
  to?: AppRoute;
```
cast a type where you need `as AppRoute`